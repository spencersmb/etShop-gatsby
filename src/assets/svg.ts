
// import CreditCard from './svg-icons/genericCreditCard.svg'
const CreditCard = require('./svg-icons/genericCreditCard.svg')
const Pinterest = require('./svg-icons/pinterest.svg')
export const svgs = {
	CreditCard,
	Pinterest
}