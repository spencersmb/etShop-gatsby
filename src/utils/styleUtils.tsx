import React, { ReactNode } from 'react'

/**
 * renderSvg(svg)
 * - Component Helper to render and svg item
 *
 * @param {Object} Svg - from importing an svg into react
 */

export const renderSvg = (Svg: string): ReactNode=> {
	if(!Svg){
		return (<div>No SVG found</div>)
	}
	return <Svg />
}
