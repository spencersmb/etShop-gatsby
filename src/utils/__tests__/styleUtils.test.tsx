import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'
import { svgs } from '../../assets/svg'
import { renderSvg } from '../styleUtils'

afterEach(cleanup)
const Parent = () => {
	return(
		<div data-testid='svg-test'>
			{renderSvg(svgs.CreditCard)}
		</div>
	)
}

describe('Login Modal', () => {
	it('Should render an SVG component', () => {
		const rendered = render(<Parent/>)
		const wrapper = rendered.getByTestId('svg-test')
		expect(wrapper.children.length).toBe(1)
	})
	it('renders SVG correctly', () => {
		const tree = renderer
			.create(<Parent/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})