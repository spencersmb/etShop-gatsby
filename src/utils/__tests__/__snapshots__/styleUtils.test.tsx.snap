// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Modal renders SVG correctly 1`] = `
<div
  data-testid="svg-test"
>
  <test-file-stub />
</div>
`;
